### PR TITLE
Implements Rados Snaps API

### PR DESCRIPTION
**Pull request by bbengfort**

### CODE REVIEW

File: rados/rados_test.go [hunk 1]
```diff
@@ -12,7 +12,7 @@ import (
 	"testing"
 	"time"
 
-	"github.com/ceph/go-ceph/rados"
+	"github.com/immesys/go-ceph/rados"
```
Comment (immesys) [inline rados/rados_test.go]: modules would make this much easier

File: rados/snapshot.go [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+
+	if ret == 0 {
+		value := C.GoString((*C.char)(unsafe.Pointer(&buf[0])))
+		return value, nil
+	}
+
+	return "", GetRadosError(ret)
+}
+
+// SnapGetStamp finds when a pool snapshort occurred
```
Comment (immesys) [inline rados/snapshot.go]: snapshort

File: rados/snapshot.go [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+	var nsec int64
+	ret := int(C.rados_ioctx_snap_get_stamp(
+		ioctx.ioctx, C.rados_snap_t(snap), (*C.time_t)(unsafe.Pointer(&nsec)),
+	))
+
+	if ret == 0 {
+		// Handle success case
+		return time.Unix(0, nsec), nil
+	}
+	return time.Time{}, GetRadosError(ret)
```
Comment (immesys) [inline rados/snapshot.go:133]: ah, the old "what to return on error with time struct" problem

File: rados/snapshot_test.go [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+)
+
+func TestSnapshots(t *testing.T) {
+	conn, _ := rados.NewConn()
+	conn.ReadDefaultConfigFile()
+	conn.Connect()
+	defer conn.Shutdown()
+
+	pool_name := GetUUID()
+	err := conn.MakePool(pool_name)
```
Comment (immesys) [inline rados/snapshot_test.go:19]: how does this work? What rule does it use? what placement groups? Interesting.

BTW you can test this on devmachine if you use dvcephgetconf

**Review by immesys — APPROVED** [review]
Looks perfect. One typo

File: rados/snapshot_test.go (hunk 4 above)
Comment (bbengfort) [reply rados/snapshot_test.go:19]: yeh, zero idea - it's what the other tests were doing … but clearly they have some other setup that I don't have. Although it did pass on stratus ceph - do you think I messed something up?

File: rados/snapshot.go (hunk 3 above)
Comment (bbengfort) [reply rados/snapshot.go:133]: so true

File: rados/snapshot.go (hunk 2 above)
Comment (bbengfort) [reply rados/snapshot.go]: the opposite of snaplong

File: rados/rados_test.go (hunk 1 above)
Comment (bbengfort) [reply rados/rados_test.go]: this is annoying - changing it back before merging

**Pull request merged**